lib/saluki-io/src/deser/codec/dogstatsd/mod.rs

```diff
@@ -450,6 +465,8 @@ fn parse_dogstatsd_service_check<'a>(
     let mut maybe_hostname = None;
     let mut maybe_tags = None;
     let mut maybe_message = None;
+    let mut maybe_container_id = None;
+    let mut maybe_external_data = None;
     let mut seen_message = false;
```

Comment:
We also need to update our decoding logic to allow for fields _after_ the message.

Specifically, the reason I noticed us failing to parse service checks with `c:` was because they came _after_ the message field... otherwise we would have just been ignoring it since it was an unknown field prefix.

We'll also need a new unit test that ensures we allow for that, too.